src/recipe.h

```diff
@@ -186,4 +210,38 @@ class recipe
         bool bp_autocalc = false;
 };
 
+template<typename _FIter>
+std::string recipe::required_skills_string( _FIter first, _FIter last, const Character *c,
+        const bool print_skill_level ) const
+{
+    if( first == last ) {
+        return _( "<color_cyan>none</color>" );
+    }
+
+    return enumerate_as_string( first, last,
+    [&]( const std::pair<skill_id, int> &skill ) {
+        const int player_skill = c ? c->get_skill_level( skill.first ) : 0;
+        std::string difficulty_color = skill.second > player_skill ? "yellow" : "green";
+        std::string skill_level_string = print_skill_level ? "" : ( std::to_string( player_skill ) + "/" );
+        skill_level_string += std::to_string( skill.second );
+        return string_format( "<color_cyan>%s</color> <color_%s>(%s)</color>",
+                              skill.first.obj().name(), difficulty_color, skill_level_string );
+    } );
+}
+
+template<typename _FIter>
+std::string recipe::required_skills_string( _FIter first, _FIter last )
```

Comment:
Do those functions really need to be in the header? They drag in several other headers.

Reply:
I moved them from the header into the cpp.